src/components/LibraryUnit.tsx

```diff
@@ -36,19 +36,24 @@ export const LibraryUnit = ({
     if (!elementsToRender) {
       return;
     }
-    const svg = exportToSvg(elementsToRender, {
-      exportBackground: false,
-      viewBackgroundColor: oc.white,
-    });
-    for (const child of ref.current!.children) {
-      if (child.tagName !== "svg") {
-        continue;
+    let svg: SVGSVGElement;
+    const current = ref.current!;
+
+    const getSvg = async () => {
+      svg = await exportToSvg(elementsToRender, {
+        exportBackground: false,
+        viewBackgroundColor: oc.white,
+      });
+      for (const child of ref.current!.children) {
+        if (child.tagName !== "svg") {
+          continue;
+        }
+        ref.current!.removeChild(child);
       }
-      ref.current!.removeChild(child);
-    }
-    ref.current!.appendChild(svg);
+      ref.current!.appendChild(svg);
+    };
 
-    const current = ref.current!;
+    getSvg();
```

Comment:
I'd instead either do a regular `promise.then`, or use an IIFE.

Reply:
> I'd instead either do a regular promise. then, or use an IIFE.

I think its ok to use any of the approaches here unless you see any pros/cons of one over the other?

Reply:
It comes to personal preference, I just thought that storing a function into a variable just to immediately call it adds unnecessary complexity.